llvm/tools/llvm-shlib/CMakeLists.txt

```diff
@@ -49,18 +46,27 @@ if(LLVM_BUILD_LLVM_DYLIB)
     ${CMAKE_CURRENT_SOURCE_DIR}/simple_version_script.map.in
     ${LLVM_LIBRARY_DIR}/tools/llvm-shlib/simple_version_script.map)
 
-    # GNU ld doesn't resolve symbols in the version script.
-    set(LIB_NAMES -Wl,--whole-archive ${LIB_NAMES} -Wl,--no-whole-archive)
-    if (NOT LLVM_LINKER_IS_SOLARISLD AND NOT MINGW)
-      # Solaris ld does not accept global: *; so there is no way to version *all* global symbols
-      set(LIB_NAMES -Wl,--version-script,${LLVM_LIBRARY_DIR}/tools/llvm-shlib/simple_version_script.map ${LIB_NAMES})
-    endif()
-    if (NOT MINGW AND NOT LLVM_LINKER_IS_SOLARISLD_ILLUMOS)
-      # Optimize function calls for default visibility definitions to avoid PLT and
-      # reduce dynamic relocations.
-      # Note: for -fno-pic default, the address of a function may be different from
-      # inside and outside libLLVM.so.
-      target_link_options(LLVM PRIVATE LINKER:-Bsymbolic-functions)
+    if(MSVC)
+      target_link_directories(LLVM PRIVATE ${LLVM_LIBRARY_DIR})
+      foreach(library ${LIB_NAMES})
+        # FIXME figure out how to use LINK_LIBRARY:WHOLE_ARCHIVE without cmake errors
```

Comment:
It would be good to handle this before landing. If we can't use the cmake feature, file a bug in their tracker and link to it from the comments.

Reply:
My guess is it could be from weird interaction from object file targets llvm uses in cmake to build both static and shared libraries.

Reply:
this was the error 
```
1> [CMake] CMake Error at cmake/modules/AddLLVM.cmake:642 (add_library):
1> [CMake]   Impossible to link target 'LLVM' because the link item 'LLVMSupport',
1> [CMake]   specified without any feature or 'DEFAULT' feature, has already occurred
1> [CMake]   with the feature 'WHOLE_ARCHIVE', which is not allowed.
```

Reply:
Not yet, that feature was introduced in CMake 3.24 but the minimum version currently supported by LLVM is 3.20.